operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/workflow/DefaultWorkflow.java

```diff
@@ -79,6 +79,9 @@ private Map<String, DependentResourceNode> toMap(Set<DependentResourceNode> node
       }
       map.put(node.getName(), node);
     }
+    if (topLevelResources.size() == 0) {
+            throw new IllegalStateException("Top level dependent resources set size must be bigger than 0 .");
```

Comment:
A more explicit message would be better as this doesn't really help the user fix the problem.